test/int32.test.js

```diff
@@ -174,6 +386,53 @@ describe('Int32', function() {
     });
   });
 
+  describe('custom casters', () => {
+    const defaultCast = mongoose.Schema.Types.Int32.cast();
+
+    afterEach(() => {
+      mongoose.Schema.Types.Int32.cast(defaultCast);
+    });
+
+    it('supports cast disabled', () => {
+      mongoose.Schema.Types.Int32.cast(false);
+      const schema = new Schema({
+        myInt1: {
+          type: Schema.Types.Int32
+        },
+        myInt2: {
+          type: Schema.Types.Int32
+        }
+      });
+      const Test = mongoose.model('Test', schema);
+      const doc = new Test({
+        myInt: '52',
+        myInt2: 52
+      });
+      assert.strictEqual(doc.myInt1, undefined);
```

Comment:
How is this test passing? from the logic in the default caster, i'd expect it to throw an error because `typeof <string>` is not equal to `number`

Reply:
By default Schema setters will set the value to undefined and set an error, I can add a check for the error, too.